homeassistant/components/switch/smappee.py

```diff
@@ -0,0 +1,99 @@
+"""
+Support for interacting with Smappee Comport Plugs.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/switch.smappee/
+"""
+import logging
+
+from homeassistant.components.smappee import DATA_SMAPPEE
+from homeassistant.components.switch import (SwitchDevice)
+
+DEPENDENCIES = ['smappee']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Comfort Plug'
+ICON = 'mdi:power-plug'
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Smappee Comfort Plugs."""
+    smappee = hass.data[DATA_SMAPPEE]
+
+    dev = []
+    if smappee.is_remote_active:
+        for location_id in smappee.locations.keys():
+            for items in smappee.info[location_id].get('actuators'):
+                if items.get('name') != '':
+                    dev.append(SmappeeSwitch(smappee,
+                                             DEFAULT_NAME,
+                                             location_id,
+                                             items.get('id')))
+
+    # Skip local switches if remote available
+    if not smappee.is_remote_active:
+        local_devices = smappee.local_devices
+        if smappee.is_local_active:
+            for items in local_devices:
+                dev.append(SmappeeSwitch(smappee,
+                                         items.get('value'),
+                                         None,
+                                         items.get('key')))
+    add_devices(dev)
+
+
+class SmappeeSwitch(SwitchDevice):
+    """Representation of a Smappee Comport Plug."""
+
+    def __init__(self, smappee, name, location_id, switch_id):
+        """Initialize a new Smappee Comfort Plug."""
+        self._name = name
+        self._state = False
+        self._smappee = smappee
+        self._location_id = location_id
+        self._switch_id = switch_id
+        self._remoteswitch = True
+        if location_id is None:
+            self._remoteswitch = False
+
+    @property
+    def name(self):
+        """Return the name of the switch."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return true if switch is on."""
+        return self._state
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend."""
+        return ICON
+
+    def turn_on(self, **kwargs):
+        """Turn on Comport Plug."""
+        self._smappee.actuator_on(self._location_id, self._switch_id,
+                                  self._remoteswitch)
+        self._smappee.actuator_on(self._location_id, self._switch_id,
+                                  self._remoteswitch)
+        self._state = True
```

Comment:
Why are you setting state to True? You can only do that if you check the command result and make sure the command was successful.

Also, Home Assistant calls your update method after calling a service, which will also query your device

Reply:
The plugs don't have any way of communicating with the rest of the smappee system, they work by just sending a signal out and then hoping it is received.

It's not really very nice so setting the state seemed to be the only way I could handle them, if theres a better way to handle switches where you can't get the status just point me in that direction and I'll look into it